apps/dav/lib/CalDAV/TimeZoneFactory.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\DAV\CalDAV;
+
+use DateTimeZone;
+
+class TimeZoneFactory {
+
+	/**
+	 * conversion table of Microsoft time zones to IANA time zones
+	 *
+	 * @var array $ms2iana
```

Comment:
```suggestion
	 * @var array<string, string> $ms2iana
```

didn't check but please make the type specific